netgrasp/netgrasp.py

```diff
@@ -1593,94 +1574,69 @@ def send_email_alerts():
-        http = httplib.HTTPConnection("api.macvendors.com", 80)
-        url = """/%s""" % mac
-        http.request("GET", url)
-        response = http.getresponse()
-
-        if response.status == 200 and response.reason == "OK":
-            vendor = response.read()
-            ng.debugger.debug("identified %s as %s", (mac, vendor))
+        source = ng.maclookup["source"]
+        if source == "local":
+            vendor = libmac.lookup_local(mac, ng.database["filename"], ng.database["lock"], ng.debugger)
+        elif source == "remote":
+            vendor = libmac.lookup_remote(mac, ng.debugger)
         else:
-            vendor = None
-            ng.debugger.info("failed to identify %s", (mac,))
-
-        return vendor
-
+            ng.debugger.error("invalid mac lookup source '%s'" % source)
     except Exception:
         ng.debugger.dump_exception("mac_lookup() caught exception")
 
+    return vendor
+
 
 def refresh_dns_cache():
     ng = netgrasp_instance
+    ng.debugger.debug("entering refresh_dns_cache()")
 
     # @TODO consider retrieving actual TTL from DNS -- for now refresh active devices regularly
     try:
-        ng.debugger.debug("entering refresh_dns_cache")
-
         ttl = datetime.datetime.now() - datetime.timedelta(minutes=15)
         ng.db.cursor.execute("SELECT host.hid, host.name, activity.did, mac.address, ip.address FROM activity LEFT JOIN ip ON activity.iid = ip.iid LEFT JOIN host ON activity.iid = host.iid LEFT JOIN mac ON ip.mid = mac.mid WHERE activity.active = 1 AND host.updated < ? LIMIT 10", (ttl,))
         rows = ng.db.cursor.fetchall()
         for row in rows:
             hid, old_name, did, mac, ip = row
             name = dns_lookup(ip)
             now = datetime.datetime.now()
-            with exclusive_lock.ExclusiveFileLock(ng, 5, "refresh_dns_cache"):
+            with exclusive_lock.ExclusiveFileLock(ng.database["lock"], 5, "refresh_dns_cache", ng.debugger):
                 ng.debugger.debug("Refreshing hostname from '%s' to '%s' for %s", (old_name, name, ip))
                 ng.db.cursor.execute("UPDATE host SET name = ?, updated = ? WHERE hid = ?", (name, now, hid))
                 ng.db.connection.commit()
-
     except Exception:
         ng.debugger.dump_exception("refresh_dns_cache() caught exception")
 
 
 def dns_lookup(ip):
+    import socket
+
     ng = netgrasp_instance
+    ng.debugger.debug("entering dns_lookup(%s)", (ip,))
 
     try:
-        import socket
-
-        ng.debugger.debug("entering dns_lookup(%s)", (ip,))
         try:
-            host_name, aliaslist, ipaddrlist = socket.gethostbyaddr(ip)
-            ng.debugger.debug("host_name(%s), aliaslist(%s), ipaddrlist(%s)", (host_name, aliaslist, ipaddrlist))
-            return host_name
-
+            hostname, aliaslist, ipaddrlist = socket.gethostbyaddr(ip)
+            ng.debugger.debug("hostname(%s), aliaslist(%s), ipaddrlist(%s)", (hostname, aliaslist, ipaddrlist))
+            return hostname
```

Comment:
again: unnecessary/unrelated cleanup swelling this PR up and making it difficult to properly review the functional changes